csrc/python_frontend/fusion_record.h

```diff
@@ -586,13 +586,11 @@ struct DimsOpRecord : RecordFunctor {
       auto arg =
           fd.getFusionState(args_.at(0).index)->template as<TensorView>();
       auto output = set(arg);
-      int rank = static_cast<int>(dims_.size());
-      std::vector<IterDomain*> allocation_domain(rank);
-      for (int i : c10::irange(rank)) {
-        allocation_domain[rank - 1 - static_cast<int>(dims_[i])] =
-            output->axis(i);
-      }
-      output->setAllocationDomain(allocation_domain, true);
+      std::vector<IterDomain*> allocation_domain = ir_utils::strideOrderToAllocation(
```

Comment:
Changing this here as well, even though `set` will remove any reduction axis for generality. The `contiguity` value is fixed.